### PR TITLE
refactor: ncg observer

### PR DESCRIPTION
**Pull request by Junyong-Suh**
- Refactored `nine-chronicles.ts` as a prior task to applying fee based on last 24hr transfer amount

### CODE REVIEW

File: bridge/test/observers/__snapshots__/nine-chronicles.spec.ts.snap [hunk 1]
```diff
@@ -27,17 +27,6 @@ Array [
       "sender": "0x2734048eC2892d111b4fbAB224400847544FC872",
     },
   ],
-  Array [
```
Comment (Junyong-Suh) [inline bridge/test/observers/__snapshots__/nine-chronicles.spec.ts.snap:30]: wondering why it is removed / is okay to be removed

File: bridge/test/observers/__snapshots__/nine-chronicles.spec.ts.snap [hunk 2]
```diff
@@ -119,44 +108,6 @@ Array [
       "text": "NCG refund event occurred.",
     },
   ],
-  Array [
```
Comment (Junyong-Suh) [inline bridge/test/observers/__snapshots__/nine-chronicles.spec.ts.snap:122]: wondering why it is removed / is okay to be removed

File: bridge/src/observers/nine-chronicles.ts [hunk 3]
```diff
@@ -118,435 +119,591 @@ export class NCGTransferredEventObserver
...
-                            cause: `Overflowed Amount ${limitedAmount.toString()} is lower than minimum NCG. Refund NCG.`,
-                            libplanetTxId: txId,
-                            sender: sender,
-                            recipient: recipient,
-                            amount: amount.toNumber(),
-                        });
-                        console.log(
-                            `Overflowed Amount after refund ${limitedAmount.toString()} is lower than minimum NCG. Refund NCG. The transaction's id is`,
-                            smallAmountRefundTxId
+                        // refund if the remaining amount is below policy minimum
```
Comment (shkangr) [inline bridge/src/observers/nine-chronicles.ts:251]: @Junyong-Suh 
I think the “await” keyword is needed in that part.

Comment (Junyong-Suh) [reply bridge/src/observers/nine-chronicles.ts:251]: @shkangr thanks for the catch. fixed that 👍

**Review by shkangr — APPROVED** [review]

**Pull request merged**